fix #42 - overwrite body and name as well

regevbr opened the pull request.

regevbr commented:
@svenstaro can you please CR?

svenstaro commented:
Hey, sorry for the super long wait. Could you rebase this? I'll review quickly after.

regevbr commented:
> Hey, sorry for the super long wait. Could you rebase this? I'll review quickly after.

Done

regevbr commented:
Noticed that the CI is failing, so I had to remove the make_latest feature as it is no longer supported by github api

regevbr commented:
Sorry for the mess - but I installed the latest version of octokit and fixed make_latest to work with it.
I also added test for promote. The test is not that stable due to GitHub API so I added retries but now everything works

regevbr commented:
@svenstaro not sure why the CI fails, do you have any idea?

svenstaro commented on README.md:
```diff
@@ -59,6 +60,7 @@ jobs:
         asset_name: mything
         tag: ${{ github.ref }}
         overwrite: true
+        promote: true
```
Do we need this here? After all, in this scenario in the README, the user won't have a prerelease here. If you want, you can add a new scenario where you first create a prerelease and then promote it.

regevbr replied:
In my case yes, the way I use this action ti to call it once a new prerelease is created manually form the github website, so I don't really use the action with prerelease first. Does it make sense? 
In any case I can jut remove if from this example, it doesn't really matter

svenstaro replied:
It's not really required or does it do anything in these examples so let's remove flags that aren't required as otherwise people might think that they need these somehow.

regevbr replied:
done

svenstaro reviewed (changes requested):
Is there any chance you could make the `promote` feature a separate PR? I think it would be much tidier.

regevbr commented:
> Is there any chance you could make the `promote` feature a separate PR? I think it would be much tidier.

sure, sperate what? the dependencies bump and ci fix?

regevbr commented on action.yml:
```diff
@@ -20,6 +20,8 @@ inputs:
   overwrite:
     description: 'Overwrite the release in case it already exists.'
+  promote:
+    description: 'Promote a prerelease to release'
```
Done

svenstaro commented:
>> Is there any chance you could make the promote feature a separate PR? I think it would be much tidier.

> sure, sperate what? the dependencies bump and ci fix?

Yes, and also the `promote` feature.

regevbr commented:
> > > Is there any chance you could make the promote feature a separate PR? I think it would be much tidier.
> 
> > sure, sperate what? the dependencies bump and ci fix?
> 
> Yes, and also the `promote` feature.

in #104

svenstaro commented:
Alright, could you still make a separate PR for the promote feature?

regevbr commented:
> Alright, could you still make a separate PR for the promote feature?

see #106

svenstaro commented:
Ok, the other stuff is merged. Needs a new final rebase and a new title and then we should be good and I'll cut a release.

regevbr commented:
> Ok, the other stuff is merged. Needs a new final rebase and a new title and then we should be good and I'll cut a release.

rebased

svenstaro commented on .github/workflows/ci.yml:
```diff
@@ -21,51 +21,82 @@ jobs:
+        with:
+          repo_token: ${{ secrets.GITHUB_TOKEN }}
+          file: README.md
+          asset_name: TEST.md
```
I recognize that this is the preferred formatting but in order to minimize the changes in this PR, could you perhaps use the original formatting for the time being?

regevbr replied:
done

svenstaro reviewed (changes requested):
Looking good, just some minor nits.

svenstaro commented:
Very cool! Merging as-is.
